### ISSUE TITLE
false positive for RET504 in case of globals

### ISSUE THREAD

**Issue by tlambert03** [charliermarsh/ruff#1310]
The following code results in `RET504 Unnecessary variable assignment before `return` statement`
 
```python
X: int | None = None

def get_x() -> int:
    global X
    if X is None:
        X = 1
    return X
```

however, as it's modifying a global, it's not unnecessary assignment.  (naturally, whether this is a good pattern is a separate discussion :joy: ... but it's not syntactically unnecessary).

(Haven't checked the behavior of flake8-return)

**Comment by charliermarsh** [comment]
Hah! Yeah, globals should probably be omitted from this check, since assignments are themselves side-effects.

**Comment by tlambert03** [comment]
Thank you! 🙏

**Comment by charliermarsh** [comment]
All @squiddy :)
